cloudbuild.yaml

```diff
@@ -28,6 +28,11 @@ steps:
     set -o errexit
     set -o xtrace
 
+    # Set this so that the build embeds the timestamp of
+    # the source commit it was built from, instead of the
+    # unix epoch.
+    export SOURCE_DATE_EPOCH=$(git log -1 --pretty=%ct)
```

Comment:
any ideas on handling unclean state?

Reply:
oh nevermind this is cloud build.

Reply:
Well unfortunately this doesn't handle `gcloud builds submit` very well, so my ~local test failed to WAI.

Let me see if I can make that a bit more graceful at least :-/

Reply:
Right now I see:
```
Step #0: + git log -1 --pretty=%ct
Step #0: fatal: not a git repository (or any parent up to mount point /)
Step #0: Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
Step #0: + export SOURCE_DATE_EPOCH=
```

Reply:
@loosebazooka talking to @imjasonh @wlynch about GCB's support it sounds like anything based on `.git` is going to be a bust.

Billy recommended `_COMMIT_DATE` from https://cloud.google.com/build/docs/configuring-builds/use-bash-and-bindings-in-substitutions#creating_substitutions_using_payload_bindings which will take some setting up on the GCB side (which obvs I cannot do anymore!), so lemme iterate on this a little bit and I'll circle back when this is RFAL.

Reply:
Ok, so I finally managed to get things plumbed through: 
<img width="673" alt="image" src="https://user-images.githubusercontent.com/2442466/209217552-6a8ebddf-3638-44e7-8866-420c4b3c5514.png">

Where I pass `_COMMIT_DATE=2022-12-22T12:16:08-05:00` on the CLI.  I verified this with ~just a static build (commented out other targets and signing stuff), since this stuff takes forever to run, the full command I used was:
```
gcloud builds submit --project mattmoor-chainguard --substitutions _COMMIT_DATE=$(git log -1 --date=iso-strict --pretty=%cd)
```

Reply:
Alright, I just did a fresh build (with the commit timestamp for this build) and got this: 
<img width="585" alt="image" src="https://user-images.githubusercontent.com/2442466/209224439-6d63db1f-cc57-49d9-bfe2-a6e7cca578c9.png">

I'm doing a build now without `--stamp` and `_COMMIT_DATE` to see what we get 🤞